src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -672,7 +672,15 @@ private void StartStream<TContext>(IHttpApplication<TContext> application)
             }
 
             _streams[_incomingFrame.StreamId] = _currentHeadersStream;
-            _ = _currentHeadersStream.ProcessRequestsAsync(application);
+            // Must not allow app code to block the connection handling loop.
+            System.Threading.ThreadPool.UnsafeQueueUserWorkItem(state =>
+                {
```

Comment:
Nit: In Kestrel we consistently use the same level of indentation for the `{` after a `=>`.